bst_plugin/forward/process_nst_extract_sensitivity_from_head_model.m

```diff
@@ -326,3 +234,141 @@
+                k(1,:) = max(max(sensitivity_surf(:, iwl, :))) + eps;   
+            end    
+            
+            sensitivity_surf(:,iwl,isUsedTime == 1) = log10( squeeze(sensitivity_surf(:, iwl, isUsedTime == 1)) ./ repmat(k(isUsedTime == 1),nb_nodes,1));
+            
+            mask = zeros(size(sensitivity_surf));
+            mask(:,iwl,:) = sensitivity_surf(:,iwl,:) < threshold_value;
+            sensitivity_surf(mask == 1 ) = 0;
+        end
+
+        displayUnit = 'dB';
+    else % linear
+
+        for iwl=1:nb_Wavelengths
+            mask = zeros(size(sensitivity_surf_sum));
+            mask(:,iwl) = sensitivity_surf_sum(:,iwl) < 10^(threshold_value)*max(sensitivity_surf_sum(:,iwl));
+            sensitivity_surf_sum(mask == 1) = 0;
+    
+            mask = zeros(size(sensitivity_surf));
+            mask(:,iwl,:) = sensitivity_surf(:,iwl,:) <  10^(threshold_value)*max(sensitivity_surf(:,iwl,:),[],'all');
+            sensitivity_surf(mask == 1 ) = 0;
+        end
+
+        displayUnit = 'mm';
+
+    end   
+
+    sResults               = repmat(db_template('resultsmat'), 1, 2*nb_Wavelengths);
+
+    % Save sensitivity maps
+    for iwl =  1:nb_Wavelengths
+        % Store restult as brainstrom structure 
+        sResults(iwl).Comment       = ['Sensitivities - WL' num2str(iwl)];
+        sResults(iwl).ImageGridAmp  = squeeze(sensitivity_surf(:,iwl,:));
+        sResults(iwl).Time          = time;
+
+        sResults(iwl).DisplayUnits  = displayUnit;
+        sResults(iwl).ChannelFlag   = ChannelFlag;
+        %sResults(iwl).HeadModelFile = OPTIONS.HeadModelFile;
+        sResults(iwl).HeadModelType = head_model.HeadModelType;
+        sResults(iwl).SurfaceFile   = file_short(head_model.SurfaceFile);
+        %sResults(1).History       = OPTIONS.History;
+        sResults(iwl) = bst_history('add', sResults(iwl), 'compute', 'sensitivity imported from MCXlab');
+    end
+
+    % Save summed sensitivity maps
+    for iwl =  1:nb_Wavelengths
+
+        % Store restult as brainstrom structure 
+        sResults(nb_Wavelengths + iwl).Comment       =  ['Summed sensitivities - WL' num2str(iwl)];
+        sResults(nb_Wavelengths + iwl).ImageGridAmp  = squeeze(sensitivity_surf_sum(:,iwl));
+        sResults(nb_Wavelengths + iwl).Time          = [1];
+
+        sResults(nb_Wavelengths + iwl).DisplayUnits  = displayUnit;
+        sResults(nb_Wavelengths + iwl).ChannelFlag   = ChannelFlag;
+        %sResults(iwl).HeadModelFile = OPTIONS.HeadModelFile;
+        sResults(nb_Wavelengths + iwl).HeadModelType = head_model.HeadModelType;
+        sResults(nb_Wavelengths + iwl).SurfaceFile   = file_short(head_model.SurfaceFile);
+        %sResults(1).History       = OPTIONS.History;
+        sResults(nb_Wavelengths + iwl) = bst_history('add', sResults(nb_Wavelengths + iwl), 'compute', 'Summed sensitivity imported from MCXlab');
+    end
+
+
+end
+
+function sResults = getCoverage(head_model, ChannelMat, ChannelFlag)
```

Comment:
ok; so i started refactoring this process. If you can complete this function tomorrow. You can check the function above for help. :)